File: src/main/java/pfs/lms/enquiry/domain/OperatingParameter.java
package pfs.lms.enquiry.domain;

import lombok.*;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ManyToOne;
import java.sql.Blob;
import java.time.LocalDate;

@Entity
@Setter
@Getter
@ToString
@EqualsAndHashCode
@NoArgsConstructor
public class OperatingParameter extends AbstractEntity {

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private LoanMonitor loanMonitor;

    private Integer serialNumber;
    private String month;
    private Integer year;
    private Double exportNetGeneration; //(Million Units)
    private Double plfCufActual; // (%age)
    private Double applicableTariff; // (Rs / Unit)
    private Double revenue; // (Rs in Lakhs)
    private LocalDate dateOfInvoice;
    private LocalDate dateOfPaymentReceipt;
    private Double carbonDioxideEmission; // (Tonnes)
    private Double waterSaved;
    private String remarks;
    private String designPlfCuf;
    private String actualYearlyAveragePlfCuf;
    private String documentType;
    private String documentTitle;
    private String fileReference;
}

File: src/main/ng/src/app/main/content/model/operatingParameter.ts
export class OperatingParameterModel {

    id: string;
    serialNumber: number;
    month: string;
    year: number;
    exportNetGeneration: number;
    plfCufActual: number;
    applicableTariff: number;
    revenue: number;
    dateOfInvoice: Date;
    dateOfPaymentReceipt: Date;
    carbonDiOxideEmission: number;
    waterSaved: number;
    remarks: string;
    designPlfCuf: string;
    actualYearlyAveragePlfCuf: string;
    documentType: string;
    documentTitle: string;
    fileReference: string;

    /**
     * constructor()
     * @param _operatingParameter 
     */
    constructor(_operatingParameter: any)
    {
        this.id = _operatingParameter.id || '';
        this.serialNumber = _operatingParameter.serialNumber || 0;
        this.month = _operatingParameter.month || '';
        this.year = _operatingParameter.year || 0;
        this.exportNetGeneration = _operatingParameter.exportNetGeneration || 0;
        this.plfCufActual = _operatingParameter.plfCufActual || 0;
        this.applicableTariff = _operatingParameter.applicableTariff || 0;
        this.revenue = _operatingParameter.revenue || 0;
        this.dateOfInvoice = _operatingParameter.dateOfInvoice || undefined;
        this.dateOfPaymentReceipt = _operatingParameter.dateOfPaymentReceipt || undefined;
        this.carbonDiOxideEmission = _operatingParameter.carbonDiOxideEmission || 0;
        this.waterSaved = _operatingParameter.waterSaved || 0;
        this.remarks = _operatingParameter.remarks || '';
        this.designPlfCuf = _operatingParameter.designPlfCuf || '';
        this.actualYearlyAveragePlfCuf = _operatingParameter.actualYearlyAveragePlfCuf || '';
        this.documentType = _operatingParameter.documentType || '';
        this.documentTitle = _operatingParameter.documentTitle || '';
        this.fileReference = _operatingParameter.fileReference || '';
    }
}
